decomp2dbg/clients/gdb/gdb_client.py

```diff
@@ -241,15 +241,18 @@ def _handle_cmd(self, args):
         handler = getattr(self, handler_str)
         handler(args)
 
+    def _is_arg_set(self, arg):
+        return arg != None
+
     def _handle_connect(self, args):
         if not args.decompiler_name:
             err("You must provide a decompiler name when using this command!")
             return
 
-        if (args.base_addr_start and not args.base_addr_end) or (args.base_addr_end and not args.base_addr_start):
+        if (self._is_arg_set(args.base_addr_start) and not self._is_arg_set(args.base_addr_end)) or (self._is_arg_set(args.base_addr_end) and not self._is_arg_set(args.base_addr_start)):
```

Comment:
This is indeed a bug, but I wouldn't say I like introducing new functions where we don't have to. Can you inline your `_is_arg_set` to all the places you used it?